projects/cm_mcu/MonitorI2CTask.c

```diff
@@ -84,8 +85,11 @@ void MonitorI2CTask(void *parameters)
   for (;;) {
     log_debug(LOG_MONI2C, "%s: grab semaphore\r\n", args->name);
     // grab the semaphore to ensure unique access to I2C controller
-    while (xSemaphoreTake(args->xSem, (TickType_t)10) == pdFALSE)
-      ;
+    if (acquireI2CSemaphore(args->xSem) == pdFAIL) {
+      log_warn(LOG_SERVICE, "%s could not get semaphore in time; continue\r\n", args->name);
+      continue;
```

Comment:
so to be clear I'm not suggesting changing this line from `continue` to `break`; I was suggesting turning the loop into something like
```
while (acquireI2CSemaphore(args->xSem) == pdFAIL) 
   ;
```
What you have here is actually ok in this instance (your `continue` will work too) and this was not one of the instances that I had commented on... In the `InitTask` the code is different.

Reply:
@pwittich Thank you so much for looking into this. In the latest commit, I have added this infinite loop attempting to access a semaphore in InitTask.c **only**, and the rest stays the way they are (i.e. either `continue` or `return`). 

I was also trying to test if the code in InitTask here works as expected by increasing the priority of CLI task and lowering the priority of InitTask so that I can do `semaphore 2 take` before it reaches this part in InitTask. However, it didn't seem to let me interact with the CLI prior to executing InitTask. If you have any suggestions on how to test this rare occasion of CLI taking a semaphore before InitTask, let me know.

Reply:
ok. can you mark all the issues that you see as resolved above as resolved?

Reply:
tbh I don't expect the CLI to beat the `InitTask` -- is that a big issue?

Reply:
I guess if we put the priority of the CLI above the monitoring tasks then it should beat other waiting tasks to acquire the semaphore, but it will still have to wait if another task has already acquired the semaphore. I think that is ok. We can tune the timeout time in `acquireSemaphore` if the CLI times out too much; I would think 5-10 seconds is probably ok?